components/centraldashboard/app/metrics_service_factory.ts

```diff
@@ -0,0 +1,33 @@
+import {MetricServiceClient} from '@google-cloud/monitoring';
+
+import {KubernetesService} from './k8s_service';
+import {MetricsService} from './metrics_service';
+import {StackdriverMetricsService} from './stackdriver_metrics_service';
+
+/**
+ * Factory function to return an instance of the MetricsService based on
+ * platform information.
+ * @returns MetricServiceClient or null if the platform is not currently
+ *  supported.
+ */
+export async function getMetricsService(k8sService: KubernetesService):
+    Promise<MetricsService>|null {
+  try {
+    const [platformInfo, nodes] = await Promise.all(
```

Comment:
```suggestion
    const [platformInfo, nodes] = await Promise.all([
        k8sService.getPlatformInfo(),
        k8sService.getNodes()
    ]);
```

 I think that's a more standard way of writing that

Reply:
Done.